### PR TITLE
Add course review rejection reason

### LINKED ISSUE #420: [BACKLOG] Add course review rejection reason (and notification)
### Problem

Currently, we don't have a way to notify the user that we rejected their course review (and why so).

### Task Description

1. Edit the schema to allow us to specify why we rejected his/her course review.
2. Display that comment on the course detail page in his/her review.
3. (BONUS) If possible, find a way to explicitly notify the user that we rejected their review (maybe through email? need investigation)

### Additional Context

This is a full stack task, primarily backend (database and api), but the frontend part should be easy (just display some more texts), so should be fine as a 1 person task.

### Related Teams

- [X] Frontend
- [X] Backend
- [ ] Data
- [ ] Design
- [ ] Infra
- [ ] QA

### Task Advisors

@bombnp

### PR DESCRIPTION
**Pull request by tithanayut**
## Why did you create this PR

- Add course review rejection reason (Closes #420)

image

## What did you do

- Update Mongoose schema, GraphQL schema, and edit `web` to display rejection reason

## Demo

[https://dev.cugetreg.com](https://dev.cugetreg.com)

## Checklist

- [x] Deploy a demo
- [x] Check browsers compatibility
- [x] Wrote coverage tests

### CODE REVIEW

File: apps/api/src/review/review.resolver.ts [hunk 1]
```diff
@@ -84,8 +84,9 @@ export class ReviewResolver {
   @Mutation('setReviewStatus')
   async setStatus(
     @Args('reviewId') reviewId: string,
-    @Args('status') status: ReviewStatus
+    @Args('status') status: ReviewStatus,
+    @Args('rejectionReason') rejectionReason?: ReviewStatus
```
Comment (bombnp) [inline apps/api/src/review/review.resolver.ts:88]: shouldn't this be string?

**Review by bombnp — CHANGES_REQUESTED** [review]
what the fuck so fast

File: apps/api/src/review/review.resolver.ts (hunk 1 above)
Comment (tithanayut) [reply apps/api/src/review/review.resolver.ts:88]: oops

**Comment by tithanayut** [comment]
Because some project cancelled the meeting so I have time!

**Comment by bombnp** [comment]
I'm trying to get it to work with Appsmith on dev env, but it looks like our CI/CD is DEAD!
image

**Comment by bombnp** [comment]
@tithanayut can u trigger a deployment of web and api on dev?

**Comment by tithanayut** [comment]
@bombnp Done

**Comment by bombnp** [comment]
image
yea it works, good stuff

**Review by bombnp — APPROVED** [review]
I already edited the appsmith to add support for this in [Dev subpage](https://appsmith.internal.cugetreg.com/applications/6155d8914a99086ac822a925/pages/61817189072a84335bf197df). You can try it there.

Nice kub.

**Pull request merged**